src/travertino/node.py

```diff
@@ -44,12 +44,12 @@ def applicator(self):
 
     @applicator.setter
     def applicator(self, applicator):
-        self._applicator = applicator
-        self.style._applicator = applicator
-
         if applicator:
             applicator.node = self
 
+        self._applicator = applicator
+        self.style._applicator = applicator
```

Comment:
Since we're thinking about the edge cases: if there's an existing applicator, should we be clearing the node reference on that applicator? I don't think it will make any difference in practice right now, but for symmetry, it couldn't hurt.

Reply:
Also - given this bit us, it would be worth adding a quick comment explaining why the order is significant.

Reply:
I hadn't thought of that, that's a good point.